src/main/java/org/opensearch/ad/model/DetectorValidationIssueType.java

```diff
@@ -25,8 +25,9 @@
     WINDOW_DELAY(AnomalyDetector.WINDOW_DELAY_FIELD),
     GENERAL_SETTINGS(AnomalyDetector.GENERAL_SETTINGS),
     RESULT_INDEX(AnomalyDetector.RESULT_INDEX_FIELD),
-    MODEL_VALIDATION_ISSUE(AnomalyDetector.MODEL_VALIDATION_ISSUE); // this is a unique case where aggregation failed but
-                                                                    // don't want to throw exception
+    TIMEOUT(AnomalyDetector.TIMEOUT),
+    AGGREGATION(AnomalyDetector.AGGREGATION); // this is a unique case where aggregation failed but
```

Comment:
Confused on what this issue type means exactly. What aggregation are you saying is failing here? Can you make the comment more clear? Why would we not want to throw an exception?

Reply:
This would happen when aggregations returns null indicating that there might be a bug or some opensearch core changes. We do this when aggregating for preview api and one or two others I believe. This comment explains why we just return empty response instead of exception in `SearchFeatureDao`(https://github.com/opensearch-project/anomaly-detection/blob/main/src%2Fmain%2Fjava%2Forg%2Fopensearch%2Fad%2Ffeature%2FSearchFeatureDao.java#L405-L407) and since an empty response means no issue in validation API instead I return that there was an aggregation issue. I will make this more clear in comments.

Reply:
Ok, got it. If this is strictly related to a problem with core instead of the detector config causing invalid aggregations in the queries then I'm fine with this.